### PR TITLE
Hotfix 2487

### LINKED ISSUE #2487: Random test failed caused by server socket closed frequently.
**Describe the bug**
more details about the issue:
https://github.com/milvus-io/pymilvus/issues/219

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**

**Environment details**
- Hardware/Softward conditions (OS, CPU, GPU, Memory)
- Method of installation (Docker, or from source)
- Milvus version (v0.3.1, or v0.4.0)
- Milvus configuration (Settings you made in `server_config.yaml`)

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by tinkerlin**
**What type of PR is this?**

BUG

**Which issue(s) this PR fixes:**

Fixes #2487

### CODE REVIEW

File: core/src/server/init/InstanceLockCheck.cpp [hunk 1]
```diff
@@ -66,5 +73,11 @@ InstanceLockCheck::Check(const std::string& path) {
     return Status::OK();
 }
 
+void
+InstanceLockCheck::Release() {
+    auto fd = open(InstanceLockCheck::GetInstance()->lk_path.c_str(), O_RDWR | O_CREAT | O_NOFOLLOW, 0640);
```
Comment (JinHai-CN) [inline core/src/server/init/InstanceLockCheck.cpp:78]: Need to check whether the lock file is openned successfully or not.

**Review by JinHai-CN — CHANGES_REQUESTED** [review]

File: core/src/server/init/InstanceLockCheck.cpp (hunk 1 above)
Comment (tinkerlin) [reply core/src/server/init/InstanceLockCheck.cpp:78]: Fix

Comment (JinHai-CN) [reply core/src/server/init/InstanceLockCheck.cpp:78]: If fd is invalid, please also print the errno and error message, to indicate the error.

**Review by JinHai-CN — APPROVED** [review]

**Comment by JinHai-CN** [comment]
Need to cherry-pick to master

**Pull request merged**